src/App.tsx

```diff
@@ -15,36 +16,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  None = '',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+const getSortedGoods = (sortField: SortType, isReversed: boolean): string[] => {
+  const sortedGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.Alphabet:
+        return good1.localeCompare(good2);
+      case SortType.Length:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+};
+
+const reset = (
+  setSortField: React.Dispatch<React.SetStateAction<SortType>>,
+  setIsReversed: React.Dispatch<React.SetStateAction<boolean>>,
+) => {
+  setSortField(SortType.None);
+  setIsReversed(false);
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const isVisibleReset = sortField || isReversed;
```

Comment:
The `isVisibleReset` condition currently checks if either `sortField` or `isReversed` is truthy. This works, but it might be clearer to explicitly check if `sortField` is not `SortType.None` and `isReversed` is true. This would make the intent of the condition more explicit.